### PR TITLE
Re-enable browsing of `TMethodBrowsable`

### PR DESCRIPTION
**Pull request by bellenot**
Prevent potential infinite recursion and simplify part of the code. This fixes the issue #13233 and doesn't affect the already closed Jira issue [#9260](https://sft.its.cern.ch/jira/browse/ROOT-9260)

### CODE REVIEW

**Comment by olifre** [comment]
> Well, what I'm doing here i prevent the following example: The class `TVector3` has a method `Orthogonal()` returning a `TVector3`, which as a method `Orthogonal()`, and so on. It's maybe not optimal, but it's a good compromise between not browsing any `TMethodBrowsable` and having infinite recursion. That being said, feel free to propose a better solution (it's @pcanal code anyway wink). At the end of the day, IIUC, what we need is to be able to (even partially) browse methods to be able to draw something...

Yes, I see. However, while the case `Orthogonal()->Orthogonal()` is seldomly useful, other simple cases like `Orthogonal()->Px()` might be useful. 

So I do actually wonder why the infinite recursion is an issue — I'm with @pcanal 's comment from #13233 here, i.e. since the recursion is real, and given that the populating is already done lazily (i.e. triggered by the user clicking), it does not break anything technically (unless I missed something). 

In other words: Is JIRA issue [#9260](https://sft.its.cern.ch/jira/browse/ROOT-9260) actually a bug? In my opinion, this would be intended behaviour (if I click on a method which returns something not really useful from the mathematical / physics point of view, ROOT has no way to know and gives me exactly what I asked for :wink: ).

**Comment by bellenot** [comment]
OK, I'll let the final word to @pcanal. From my ( the GUI) side, I just have to remove the ` && !obj->InheritsFrom("TMethodBrowsable")` statement (that would have saved some of my time investigating all this...)

**Comment by bellenot** [comment]
P.S. @olifre Note that there are still some potential issues without my changes:
```
root [1] TBrowser b
(TBrowser &) Name: Browser Title: ROOT Object Browser
root [2] Error in <TTreeFormula::DefinedVariable>: operator is not a datamember of ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<Double32_t> >
Error in <TTreeFormula::Compile>:  Bad numerical expression : "tracks.operator"
Info in <TSelectorDraw::AbortProcess>: Variable compilation failed: {tracks.operator-().operator+().Py(),}
Error in <TTreeFormula::DefinedVariable>: operator is not a datamember of ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<Double32_t> >
Error in <TTreeFormula::Compile>:  Bad numerical expression : "tracks.operator"
Info in <TSelectorDraw::AbortProcess>: Variable compilation failed: {tracks.operator-().Vect().unit().unit().Rho(),}
```

**Comment by olifre** [comment]
> P.S. @olifre Note that there are still some potential issues without my changes:  [...]

My feeling here is that the easiest approach might be to filter out the unary operators in general, i.e. exactly the one filtering line you also added in this PR :wink:. 
Another option would be to rewrite them before passing things to `TTreeFormula`, but I'm not sure it is really worth the effort — browsing into unary operators seems to be rarely useful.

**Comment by bellenot** [comment]
@olifre done

**Comment by olifre** [comment]
> @olifre done

This LGTM :+1: , many thanks!

**Comment by bellenot** [comment]
> > @olifre done
> 
> This LGTM 👍 , many thanks!

You're very welcome!

**Review by pcanal — APPROVED** [review]
LGTM
